cluster/src/dunit/scala/io/snappydata/cluster/PreparedQueryRoutingDUnitTest.scala

```diff
@@ -908,4 +908,37 @@ class PreparedQueryRoutingDUnitTest(val s: String)
     rs1.close()
     ps3.close()
   }
+
+  def testPreparedStatementUnicodeBug(): Unit = {
+    serverHostPort = AvailablePortHelper.getRandomAvailableTCPPort
+    vm2.invoke(classOf[ClusterManagerTestBase], "startNetServer", serverHostPort)
+    logInfo(s"testPreparedStatementUnicodeBug: network server started at $serverHostPort")
+    val conn = DriverManager.getConnection(
+      "jdbc:snappydata://localhost:" + serverHostPort)
+
+    logInfo(s"testPreparedStatementUnicodeBug: Connected to $serverHostPort")
+    val stmt1 = conn.createStatement()
+
+    stmt1.execute("create table region (val string, description string) using column")
+    stmt1.execute("insert into region values ('粤' , 'unicode')")
+    stmt1.execute("insert into region values ('A', 'ascii')")
+
+    var pstmt1 = conn.prepareStatement("select * from region where val = ?")
+    pstmt1.setString(1, "A")
+    var rs1 = pstmt1.executeQuery()
+    assert(rs1.next())
+    pstmt1.close()
+
+    pstmt1 = conn.prepareStatement("select * from region where val = ?")
+    pstmt1.setString(1, "粤")
+    rs1 = pstmt1.executeQuery()
+    assert(rs1.next())
+    assert("粤" == rs1.getString(1) )
+
+    stmt1.execute("insert into region select '\u7ca5', 'unicode2'")
+    pstmt1.setString(1, "\u7ca5")
+    rs1 = pstmt1.executeQuery()
+    assert(rs1.next())
+    assert("粥" == rs1.getString(1) )
```

Comment:
shouldn't this be two rows now i.e. the last two lines can be repeated followed by "assert(!rs1.next())"

Reply:
The character differ